rest-services/server/src/main/java/com/_4point/aem/docservices/rest_services/server/assembler/AssembleDocuments.java

```diff
@@ -0,0 +1,243 @@
+					  document.createAttribute("numRequestedBlock");
+					  numRequestedBlockAttr.setValue(Integer.toString(assemblerResult.
+					  getNumRequestedBlocks()));
+					  latestBatesNum.setAttributeNode(numRequestedBlockAttr);
+					  
+					  Element successfulDocumentName =
+					  document.createElement("successfulDocumentNames");
+					  root.appendChild(successfulDocumentName);
+					  if(CollectionUtils.isNotEmpty(assemblerResult.getSuccessfulBlockNames())) {
+					  assemblerResult.getSuccessfulBlockNames().forEach(successFullDocName -> {
+					  Element successfulDoc = document.createElement("successfulDocumentName");
+					  successfulDoc.appendChild(document.createTextNode(successFullDocName));
+					  successfulDocumentName.appendChild(successfulDoc); }); }
+					  
+					  Element successfulBlocNames = document.createElement("successfulBlocNames");
+					  root.appendChild(successfulBlocNames);
+					  if(CollectionUtils.isNotEmpty(assemblerResult.getSuccessfulBlockNames())) {
+					  assemblerResult.getSuccessfulBlockNames().forEach(blocName -> { Element
+					  successfulBloc = document.createElement("successfulBlocName");
+					  successfulBloc.appendChild(document.createTextNode(blocName));
+					  successfulBlocNames.appendChild(successfulBloc); }); }
+					  
+					  Element multipleResultBloc = document.createElement("multipleResultBlocs");
+					  root.appendChild(multipleResultBloc);
+					  if(MapUtils.isNotEmpty(assemblerResult.getMultipleResultsBlocks())) {
+					  assemblerResult.getMultipleResultsBlocks().forEach((blockName,docNames)-> {
+					  Element resultBlockName = document.createElement("resultBlockName");
+					  root.appendChild(resultBlockName); Attr nameAttr =
+					  document.createAttribute("name"); nameAttr.setValue(blockName);
+					  resultBlockName.setAttributeNode(nameAttr); Element documentNames =
+					  document.createElement("documentNames"); root.appendChild(documentNames);
+					  
+					  }); }
+					  
+					  Element joblog = document.createElement("jobLog"); root.appendChild(joblog);
+					  if(assemblerResult.getJobLog() !=null) { Attr logAttr =
+					  document.createAttribute("joblogValue"); try {
+					  logAttr.setValue(Base64.getEncoder().encodeToString(assemblerResult.getJobLog
+					  ().getInlineData())); } catch (DOMException | IOException e) {
+					  log.error("Error in converting jobLog to bas64 String"); }
+					  joblog.setAttributeNode(logAttr); joblog.setAttributeNode(logAttr);
+					  }				 
+				});
+			}
+			DOMSource domSource = new DOMSource(document);
+			Transformer transformer = TransformerFactory.newInstance().newTransformer();
+			StringWriter sw = new StringWriter();
+			StreamResult sr = new StreamResult(sw);
+			transformer.transform(domSource, sr);
+			return sw.toString();
+		} catch (Exception e) {
+			throw new AssemblerServiceException("Error while converting assemblerResult to xml ", e);
+		}
```

Comment:
I have added changes as per my understanding from your comment...But I am still not sure about the IoException which I handled in convertAssemblerResultToxml() method. I am unable to rethrow it so I have logged it in catch block with exception message,

Reply:
I think we had a small misunderstanding here.  It's OK to catch exceptions, just not `java.lang.Exception`.  `java.lang.Exception` is the base class for all exceptions.  If you catch`java.lang.Exception` then you will catch everything - runtime exceptions, checked exceptions, everything.  That almost always too much.  You should try and catch more specific exceptions than that.  They've even made this easier, by creating a multi-exception catch clause so that you can be specific about what exceptions you want to catch.

Also, the goal of the try/catch mechanism is to try and move the error handling code out-of-line from the main logic so that it doesn't disrupt the flow of the logic when someone is reading a program.  If you find yourself placing multiple try/catches around several small blocks within a routine, that's not ideal.  Ideally, you should prefer to surround a larger block of the code with a try/multi-catch instead.  This isn't a hard and fast rule.  try-with-resources sometimes makes it hard to do this, however breaking the code that you want to surround with a try catch into its own subroutine is another way of attacking the problem.  I did this in your branch to show you what I mean.  Look for the toBase64String() routine.